third_party/nginx_latency_status_module/src/ngx_http_latency_stub_status_module.c

```diff
@@ -0,0 +1,305 @@
+  }
+
+  shm_zone->data = record_set;
+  return NGX_OK;
+}
+
+
+// Parse an integer input from a config. arg_index is the index of the directive
+// argument to be parsed, where the name of the directive is argument 0.
+ngx_int_t ngx_parse_int(ngx_conf_t* cf, ngx_int_t arg_index) {
+  ngx_str_t *value;
+
+  value = cf->args->elts;
+  return ngx_atoi(value[arg_index].data, value[arg_index].len);
+}
+
+
+// Parse the config for the store_latency directive.
+// It expects 3 arguments, base, scaled_factor, and max_exponent,  defining the buckets
+// for the latency distributions that will be stored.
+// There will be max_exponent + 2 buckets with the boundaries
+// [0, scaled_factor * base ^ i) for i = 0
+// [scaled_factor * base ^ (i - 1), scaled_factor * base ^ i ) for 0 < i <= max_exponent
+// [scaled_factor * base ^ (i - 1), infinity) for i = max_exponent + 1
+char* ngx_http_latency(ngx_conf_t* cf, ngx_command_t* cmd, void* conf){
+  ngx_http_latency_main_conf_t* main_conf;
+  ngx_int_t base, scale_factor, max_exponent;
+  ngx_str_t *config_value;
+
+  config_value = cf->args->elts;
+  main_conf = (ngx_http_latency_main_conf_t*) conf;
+
+  base = ngx_parse_int(cf, 1);
+  if (base == NGX_ERROR) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_stub: base \"%V\"", &config_value[1]);
+    return NGX_CONF_ERROR;
+  }
+
+  scale_factor = ngx_parse_int(cf, 2);
+  if (scale_factor == NGX_ERROR) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_stub: scaled_factor \"%V\"", &config_value[2]);
+    return NGX_CONF_ERROR;
+  }
+
+  max_exponent = ngx_parse_int(cf, 3);
+  if (max_exponent == NGX_ERROR) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_stub: max_value \"%V\"", &config_value[3]);
+    return NGX_CONF_ERROR;
+  }
+
+  if (base <= 1) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_stub: base %d. It must be greater than 1", base);
+    return NGX_CONF_ERROR;
+  }
+  if (scale_factor <= 0) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_sub: scaled_factor %d. It must be greater than 0", scale_factor);
+    return NGX_CONF_ERROR;
+  }
+  if (max_exponent < 1) {
+    ngx_conf_log_error(NGX_LOG_EMERG, cf, 0,
+                       "invalid value for latency_status_sub: max_exponent %d. It must be at least 1", max_exponent);
+    return NGX_CONF_ERROR;
+  }
+
+  main_conf->base = base;
+  main_conf->scale_factor = scale_factor;
+  main_conf->max_exponent = max_exponent;
+  shm_max_exponent = max_exponent;
+
+  ngx_int_t* bucket_bounds;
+  ngx_int_t power = 1;
+  bucket_bounds = ngx_palloc(cf->pool, sizeof(ngx_int_t) * (max_exponent + 1));
+  if (bucket_bounds == NULL) {
+    return NGX_CONF_ERROR;
+  }
+  for (ngx_int_t i = 0; i <= max_exponent; i++) {
```

Comment:
Nit: `for (ngx_int_t i = 0, ngx_int_t power = 1; i <= max_exponent; i++) {`

Reply:
done